{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Analyzing file 1 of 100: CP002479.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 55.94%\n",
      "\n",
      "\n",
      "Analyzing file 2 of 100: CP002512.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 34.16%\n",
      "\n",
      "\n",
      "Analyzing file 3 of 100: CP002514.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 56.99%\n",
      "\n",
      "\n",
      "Analyzing file 4 of 100: CP002521.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 66.15%\n",
      "\n",
      "\n",
      "Analyzing file 5 of 100: CP002528.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 33.06%\n",
      "\n",
      "\n",
      "Analyzing file 6 of 100: CP002542.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 35.43%\n",
      "\n",
      "\n",
      "Analyzing file 7 of 100: CP002543.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 33.02%\n",
      "\n",
      "\n",
      "Analyzing file 8 of 100: CP002544.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 40.21%\n",
      "\n",
      "\n",
      "Analyzing file 9 of 100: CP002546.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 55.44%\n",
      "\n",
      "\n",
      "Analyzing file 10 of 100: CP002547.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 42.30%\n",
      "\n",
      "\n",
      "Analyzing file 11 of 100: CP002548.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 21.82%\n",
      "\n",
      "\n",
      "Analyzing file 12 of 100: CP002568.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 63.36%\n",
      "\n",
      "\n",
      "Analyzing file 13 of 100: CP002584.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 36.91%\n",
      "\n",
      "\n",
      "Analyzing file 14 of 100: CP002593.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 72.65%\n",
      "\n",
      "\n",
      "Analyzing file 15 of 100: CP002606.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 35.52%\n",
      "\n",
      "\n",
      "Analyzing file 16 of 100: CP002623.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 52.01%\n",
      "\n",
      "\n",
      "Analyzing file 17 of 100: CP002628.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 56.12%\n",
      "\n",
      "\n",
      "Analyzing file 18 of 100: CP002629.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 44.56%\n",
      "\n",
      "\n",
      "Analyzing file 19 of 100: CP002630.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 61.71%\n",
      "\n",
      "\n",
      "Analyzing file 20 of 100: CP002638.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 67.62%\n",
      "\n",
      "\n",
      "Analyzing file 21 of 100: CP002657.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 64.09%\n",
      "\n",
      "\n",
      "Analyzing file 22 of 100: CP002663.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 49.46%\n",
      "\n",
      "\n",
      "Analyzing file 23 of 100: CP002666.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 75.36%\n",
      "\n",
      "\n",
      "Analyzing file 24 of 100: CP002667.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 31.67%\n",
      "\n",
      "\n",
      "Analyzing file 25 of 100: CP002683.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 39.97%\n",
      "\n",
      "\n",
      "Analyzing file 26 of 100: CP002690.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 30.60%\n",
      "\n",
      "\n",
      "Analyzing file 27 of 100: CP002691.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 45.75%\n",
      "\n",
      "\n",
      "Analyzing file 28 of 100: CP002701.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 14.74%\n",
      "\n",
      "\n",
      "Analyzing file 29 of 100: CP002725.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 34.82%\n",
      "\n",
      "\n",
      "Analyzing file 30 of 100: CP002738.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 46.95%\n",
      "\n",
      "\n",
      "Analyzing file 31 of 100: CP002786.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 59.09%\n",
      "\n",
      "\n",
      "Analyzing file 32 of 100: CP002810.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 72.57%\n",
      "\n",
      "\n",
      "Analyzing file 33 of 100: CP002825.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 26.95%\n",
      "\n",
      "\n",
      "Analyzing file 34 of 100: CP002831.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 45.68%\n",
      "\n",
      "\n",
      "Analyzing file 35 of 100: CP002835.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 38.51%\n",
      "\n",
      "\n",
      "Analyzing file 36 of 100: CP002858.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 35.63%\n",
      "\n",
      "\n",
      "Analyzing file 37 of 100: CP002859.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 44.47%\n",
      "\n",
      "\n",
      "Analyzing file 38 of 100: CP002874.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 20.55%\n",
      "\n",
      "\n",
      "Analyzing file 39 of 100: CP002876.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 39.43%\n",
      "\n",
      "\n",
      "Analyzing file 40 of 100: CP002899.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 26.57%\n",
      "\n",
      "\n",
      "Analyzing file 41 of 100: CP002901.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 52.82%\n",
      "\n",
      "\n",
      "Analyzing file 42 of 100: CP002902.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 56.37%\n",
      "\n",
      "\n",
      "Analyzing file 43 of 100: CP002930.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 44.78%\n",
      "\n",
      "\n",
      "Analyzing file 44 of 100: CP002955.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 36.04%\n",
      "\n",
      "\n",
      "Analyzing file 45 of 100: CP002959.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 53.19%\n",
      "\n",
      "\n",
      "Analyzing file 46 of 100: CP002961.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 33.30%\n",
      "\n",
      "\n",
      "Analyzing file 47 of 100: CP002976.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 56.03%\n",
      "\n",
      "\n",
      "Analyzing file 48 of 100: CP002985.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 51.89%\n",
      "\n",
      "\n",
      "Analyzing file 49 of 100: CP002987.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 32.55%\n",
      "\n",
      "\n",
      "Analyzing file 50 of 100: CP002991.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 31.79%\n",
      "\n",
      "\n",
      "Analyzing file 51 of 100: CP002999.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 39.47%\n",
      "\n",
      "\n",
      "Analyzing file 52 of 100: CP003051.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 63.74%\n",
      "\n",
      "\n",
      "Analyzing file 53 of 100: CP003058.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 43.18%\n",
      "\n",
      "\n",
      "Analyzing file 54 of 100: CP003075.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 57.04%\n",
      "\n",
      "\n",
      "Analyzing file 55 of 100: CP003096.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 44.46%\n",
      "\n",
      "\n",
      "Analyzing file 56 of 100: CP003108.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 39.35%\n",
      "\n",
      "\n",
      "Analyzing file 57 of 100: CP003119.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 64.17%\n",
      "\n",
      "\n",
      "Analyzing file 58 of 100: CP003130.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 54.63%\n",
      "\n",
      "\n",
      "Analyzing file 59 of 100: CP003147.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 45.98%\n",
      "\n",
      "\n",
      "Analyzing file 60 of 100: CP003153.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 63.25%\n",
      "\n",
      "\n",
      "Analyzing file 61 of 100: CP003154.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 60.63%\n",
      "\n",
      "\n",
      "Analyzing file 62 of 100: CP003155.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 42.84%\n",
      "\n",
      "\n",
      "Analyzing file 63 of 100: CP003156.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 38.70%\n",
      "\n",
      "\n",
      "Analyzing file 64 of 100: CP003171.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 50.84%\n",
      "\n",
      "\n",
      "Analyzing file 65 of 100: CP003178.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 42.59%\n",
      "\n",
      "\n",
      "Analyzing file 66 of 100: CP003183.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 52.25%\n",
      "\n",
      "\n",
      "Analyzing file 67 of 100: CP003191.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 38.32%\n",
      "\n",
      "\n",
      "Analyzing file 68 of 100: CP003198.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 43.75%\n",
      "\n",
      "\n",
      "Analyzing file 69 of 100: CP003203.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 58.35%\n",
      "\n",
      "\n",
      "Analyzing file 70 of 100: CP003236.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 64.99%\n",
      "\n",
      "\n",
      "Analyzing file 71 of 100: CP003255.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 56.40%\n",
      "\n",
      "\n",
      "Analyzing file 72 of 100: CP003257.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 28.03%\n",
      "\n",
      "\n",
      "Analyzing file 73 of 100: CP003260.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 37.36%\n",
      "\n",
      "\n",
      "Analyzing file 74 of 100: CP003264.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 35.68%\n",
      "\n",
      "\n",
      "Analyzing file 75 of 100: CP003273.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 39.96%\n",
      "\n",
      "\n",
      "Analyzing file 76 of 100: CP003280.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 33.98%\n",
      "\n",
      "\n",
      "Analyzing file 77 of 100: CP003281.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 34.12%\n",
      "\n",
      "\n",
      "Analyzing file 78 of 100: CP003283.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 33.94%\n",
      "\n",
      "\n",
      "Analyzing file 79 of 100: CP003313.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 32.97%\n",
      "\n",
      "\n",
      "Analyzing file 80 of 100: CP003315.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 12.42%\n",
      "\n",
      "\n",
      "Analyzing file 81 of 100: CP003345.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 27.84%\n",
      "\n",
      "\n",
      "Analyzing file 82 of 100: CP003346.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 42.67%\n",
      "\n",
      "\n",
      "Analyzing file 83 of 100: CP003349.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 35.49%\n",
      "\n",
      "\n",
      "Analyzing file 84 of 100: CP003350.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 62.74%\n",
      "\n",
      "\n",
      "Analyzing file 85 of 100: CP003359.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 30.40%\n",
      "\n",
      "\n",
      "Analyzing file 86 of 100: CP003360.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 48.12%\n",
      "\n",
      "\n",
      "Analyzing file 87 of 100: CP003364.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 60.24%\n",
      "\n",
      "\n",
      "Analyzing file 88 of 100: CP003379.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 58.96%\n",
      "\n",
      "\n",
      "Analyzing file 89 of 100: CP003382.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 61.75%\n",
      "\n",
      "\n",
      "Analyzing file 90 of 100: CP003389.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 67.09%\n",
      "\n",
      "\n",
      "Analyzing file 91 of 100: CP003390.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 42.29%\n",
      "\n",
      "\n",
      "Analyzing file 92 of 100: CP003403.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 43.47%\n",
      "\n",
      "\n",
      "Analyzing file 93 of 100: CP003415.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 42.24%\n",
      "\n",
      "\n",
      "Analyzing file 94 of 100: CP003418.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 28.56%\n",
      "\n",
      "\n",
      "Analyzing file 95 of 100: CP003422.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 52.20%\n",
      "\n",
      "\n",
      "Analyzing file 96 of 100: CP003466.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 55.85%\n",
      "\n",
      "\n",
      "Analyzing file 97 of 100: CP003470.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 66.88%\n",
      "\n",
      "\n",
      "Analyzing file 98 of 100: CP003488.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 33.83%\n",
      "\n",
      "\n",
      "Analyzing file 99 of 100: CP003493.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 65.93%\n",
      "\n",
      "\n",
      "Analyzing file 100 of 100: CP003495.embl\n",
      "\n",
      "\n",
      "Intergenomic GC content: 67.02%\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "import pandas as pd\n",
    "from os import walk\n",
    "import os\n",
    "mypath = os.path.abspath(os.getcwd()) + \"/filtered_genomes\"\n",
    "\n",
    "files = []\n",
    "for (dirpath, dirnames, filenames) in walk(mypath):\n",
    "    files.extend(filenames)\n",
    "    break\n",
    "\n",
    "ext = \"embl\"\n",
    "\n",
    "FileNames = []\n",
    "IntergenomicGC = []\n",
    "AAA = []\n",
    "ACA = []\n",
    "AGA = []\n",
    "TTA = []\n",
    "TCA = []\n",
    "CAA = []\n",
    "CTA = []\n",
    "CCA = []\n",
    "CGA = []\n",
    "GAA = []\n",
    "GTA = []\n",
    "GCA = []\n",
    "GGA = []\n",
    "AAT = []\n",
    "ACT = []\n",
    "AGT = []\n",
    "TAT = []\n",
    "TTT = []\n",
    "TCT = []\n",
    "TGT = []\n",
    "CAT = []\n",
    "CTT = []\n",
    "CCT = []\n",
    "CGT = []\n",
    "GAT = []\n",
    "GTT = []\n",
    "GCT = []\n",
    "GGT = []\n",
    "AAC = []\n",
    "ACC = []\n",
    "AGC = [] \n",
    "TAC = []\n",
    "TTC = []\n",
    "TCC = []\n",
    "TGC = []\n",
    "CAC = []\n",
    "CTC = []\n",
    "CCC = []\n",
    "CGC = []\n",
    "GAC = []\n",
    "GTC = []\n",
    "GCC = []\n",
    "GGC = []\n",
    "AAG = []\n",
    "ACG = []\n",
    "AGG = []\n",
    "TTG = []\n",
    "TCG = []\n",
    "CAG = []\n",
    "CTG = []\n",
    "CCG = []\n",
    "CGG = []\n",
    "GAG = []\n",
    "GTG = []\n",
    "GCG = []\n",
    "GGG = []\n",
    "\n",
    "\n",
    "files_tot = len(files)\n",
    "for index, name in enumerate(files):\n",
    "    print(\"\\n\")\n",
    "    print(f\"Analyzing file {index+1} of {files_tot}: {name}\")\n",
    "    print(\"\\n\")\n",
    "    record = SeqIO.read(f'filtered_genomes/{name}', ext)\n",
    "    sequence = record.seq\n",
    "    FileNames.append(str(name))\n",
    "    \n",
    "    start = 0\n",
    "    coding_sequences = []\n",
    "    intergenomic_sequence = []  # It will contain the sequences out of the locations bounds\n",
    "    intergenomic_sequence_list = []  # It will contain the locations bounds\n",
    "    intergenomic_sequence_string = \"\"  # intergenomic sequence\n",
    "    \n",
    "    for feature in record.features:\n",
    "\n",
    "        locations= feature.location\n",
    "        s = int(locations.nofuzzy_start)\n",
    "        e = int(locations.nofuzzy_end)\n",
    "        st = int(locations.strand)\n",
    "        \n",
    "        gene_start = int(locations.nofuzzy_start)\n",
    "\n",
    "        if gene_start > start:\n",
    "            ig_sequence = sequence[start:gene_start]\n",
    "            intergenomic_sequence.append(str(ig_sequence))\n",
    "            intergenomic_sequence_list.append((start, gene_start))\n",
    "\n",
    "        new_start = int(locations.nofuzzy_end)\n",
    "\n",
    "        if new_start != len(sequence):\n",
    "            start = new_start\n",
    "\n",
    "        #Reversing complementary strands\n",
    "        if st == -1:\n",
    "            sequence_negativestrand = str(sequence[s:e])\n",
    "            sequence_negativestrand= sequence_negativestrand[::-1]\n",
    "            sequence_negativestrand= sequence_negativestrand.replace(\"A\",\"b\")\n",
    "            sequence_negativestrand= sequence_negativestrand.replace(\"T\",\"A\")\n",
    "            sequence_negativestrand= sequence_negativestrand.replace(\"b\",\"T\")\n",
    "            sequence_negativestrand= sequence_negativestrand.replace(\"C\",\"d\")\n",
    "            sequence_negativestrand= sequence_negativestrand.replace(\"G\",\"C\")\n",
    "            sequence_negativestrand= sequence_negativestrand.replace(\"d\",\"G\")\n",
    "\n",
    "        #Discarding the pseudogenes\n",
    "\n",
    "            ok = 0\n",
    "            genes = str(sequence_negativestrand)\n",
    "\n",
    "\n",
    "            if len(genes)%3 != 0:\n",
    "                ok= 1\n",
    "            elif genes[:3] != \"ATG\" and genes[:3] != \"GTG\" and genes[:3] != \"TTG\":\n",
    "                ok= 2\n",
    "            elif genes[-3:] != \"TAA\" and genes[-3:] != \"TAG\" and genes[-3:] != \"TGA\":\n",
    "                ok= 3\n",
    "\n",
    "\n",
    "            if st == -1 and ok == 0:\n",
    "                coding_sequences.append(genes)\n",
    "        \n",
    "        ok = 0\n",
    "        genes = str(sequence[s:e])\n",
    "    \n",
    "        if len(genes)%3 != 0:\n",
    "            ok= 1\n",
    "        elif genes[:3] != \"ATG\" and genes[:3] != \"GTG\" and genes[:3] != \"TTG\":\n",
    "            ok= 2\n",
    "        elif genes[-3:] != \"TAA\" and genes[-3:] != \"TAG\" and genes[-3:] != \"TGA\":\n",
    "            ok= 3\n",
    "        elif len(genes) == len(sequence):\n",
    "            ok= 4\n",
    "\n",
    "        if st == 1 and ok == 0:\n",
    "            coding_sequences.append(genes)\n",
    "\n",
    "        #checking for errors\n",
    "        check = False\n",
    "        if check and st == 1 and ok != 0:\n",
    "            print(locations)\n",
    "            print(ok)\n",
    "            print(genes[:3])\n",
    "            print(genes[-3:])\n",
    "            \n",
    "    if start != len(sequence):\n",
    "        ig_sequence = sequence[start:len(sequence)]\n",
    "        intergenomic_sequence.append(str(ig_sequence))\n",
    "        intergenomic_sequence_list.append((start, gene_start))\n",
    "\n",
    "    intergenomic_sequence_string = \"\".join(intergenomic_sequence)\n",
    "\n",
    "    # Intergenomic GC content\n",
    "    IgGC = intergenomic_sequence_string.count(\"G\") + intergenomic_sequence_string.count(\"C\")\n",
    "    IgGC_content = (IgGC / len(intergenomic_sequence_string)) * 100\n",
    "    print(f\"Intergenomic GC content: {IgGC_content:.2f}%\")\n",
    "    IntergenomicGC.append(IgGC_content)\n",
    "    \n",
    "    coding_sequencewhole = \"\".join(coding_sequences)\n",
    "    aminoacids_sequence = Seq(coding_sequencewhole).translate()\n",
    "    \n",
    "    \n",
    "    #Codon counting \n",
    "    AAAcount = 0\n",
    "    AATcount = 0\n",
    "    AACcount = 0\n",
    "    AAGcount = 0\n",
    "    ACAcount = 0\n",
    "    ACCcount = 0\n",
    "    ACGcount = 0\n",
    "    ACTcount = 0\n",
    "    AGAcount = 0\n",
    "    AGTcount = 0\n",
    "    AGCcount = 0\n",
    "    AGGcount = 0\n",
    "    TATcount = 0\n",
    "    TACcount = 0\n",
    "    TTAcount = 0\n",
    "    TTTcount = 0\n",
    "    TTCcount = 0\n",
    "    TTGcount = 0\n",
    "    TCAcount = 0\n",
    "    TCCcount = 0\n",
    "    TCGcount = 0\n",
    "    TCTcount = 0\n",
    "    TGTcount = 0\n",
    "    TGCcount = 0\n",
    "    CAAcount = 0  \n",
    "    CATcount = 0\n",
    "    CACcount = 0\n",
    "    CAGcount = 0\n",
    "    CTAcount = 0\n",
    "    CTCcount = 0\n",
    "    CTGcount = 0\n",
    "    CTTcount = 0\n",
    "    CCAcount = 0\n",
    "    CCCcount = 0\n",
    "    CCGcount = 0\n",
    "    CCTcount = 0\n",
    "    CGAcount = 0\n",
    "    CGCcount = 0\n",
    "    CGGcount = 0\n",
    "    CGTcount = 0\n",
    "    GAAcount = 0\n",
    "    GATcount = 0\n",
    "    GACcount = 0\n",
    "    GAGcount = 0\n",
    "    GTAcount = 0\n",
    "    GTCcount = 0\n",
    "    GTGcount = 0\n",
    "    GTTcount = 0\n",
    "    GCAcount = 0\n",
    "    GCCcount = 0\n",
    "    GCGcount = 0\n",
    "    GCTcount = 0\n",
    "    GGTcount = 0\n",
    "    GGCcount = 0\n",
    "    GGAcount = 0\n",
    "    GGGcount = 0\n",
    "    \n",
    "    for cds in coding_sequences:\n",
    "        for i in range(0, len(cds), 3):\n",
    "            codons = cds[i: i + 3]\n",
    "            AAA1 = codons.count(\"AAA\")\n",
    "            AAAcount += AAA1\n",
    "            AAT1 = codons.count(\"AAT\")\n",
    "            AATcount += AAT1\n",
    "            AAC1 = codons.count(\"AAC\")\n",
    "            AACcount += AAC1\n",
    "            AAG1 = codons.count(\"AAG\")\n",
    "            AAGcount += AAG1\n",
    "            ACA1 = codons.count(\"ACA\")\n",
    "            ACAcount += ACA1\n",
    "            ACC1 = codons.count(\"ACC\")\n",
    "            ACCcount += ACC1\n",
    "            ACG1 = codons.count(\"ACG\")\n",
    "            ACGcount += ACG1\n",
    "            ACT1 = codons.count(\"ACT\")\n",
    "            ACTcount += ACT1\n",
    "            AGA1 = codons.count(\"AGA\")\n",
    "            AGAcount += AGA1\n",
    "            AGT1 = codons.count(\"AGT\")\n",
    "            AGTcount += AGT1\n",
    "            AGC1 = codons.count(\"AGC\")\n",
    "            AGCcount += AGC1\n",
    "            AGG1 = codons.count(\"AGG\")\n",
    "            AGGcount += AGG1\n",
    "            CAA1 = codons.count(\"CAA\")\n",
    "            CAAcount += CAA1\n",
    "            CAT1 = codons.count(\"CAT\")\n",
    "            CATcount += CAT1\n",
    "            CAC1 = codons.count(\"CAC\")\n",
    "            CACcount += CAC1\n",
    "            CAG1 = codons.count(\"CAG\")\n",
    "            CAGcount += CAG1\n",
    "            CCA1 = codons.count(\"CCA\")\n",
    "            CCAcount += CCA1\n",
    "            CCC1 = codons.count(\"CCC\")\n",
    "            CCCcount += CCC1\n",
    "            CCG1 = codons.count(\"CCG\")\n",
    "            CCGcount += CCG1\n",
    "            CCT1 = codons.count(\"CCT\")\n",
    "            CCTcount += CCT1\n",
    "            CTA1 = codons.count(\"CTA\")\n",
    "            CTAcount += CTA1\n",
    "            CTC1 = codons.count(\"CTC\")\n",
    "            CTCcount += CTC1\n",
    "            CTG1 = codons.count(\"CTG\")\n",
    "            CTGcount += CTG1\n",
    "            CTT1 = codons.count(\"CTT\")\n",
    "            CTTcount += CTT1\n",
    "            CGA1 = codons.count(\"CGA\")\n",
    "            CGAcount += CGA1\n",
    "            CGC1 = codons.count(\"CGC\")\n",
    "            CGCcount += CGC1\n",
    "            CGG1 = codons.count(\"CGG\")\n",
    "            CGGcount += CGG1\n",
    "            CGT1 = codons.count(\"CGT\")\n",
    "            CGTcount += CGT1\n",
    "            TAT1 = codons.count(\"TAT\")\n",
    "            TATcount += TAT1\n",
    "            TAC1 = codons.count(\"TAC\")\n",
    "            TACcount += TAC1\n",
    "            TCA1 = codons.count(\"TCA\")\n",
    "            TCAcount += TCA1\n",
    "            TCC1 = codons.count(\"TCC\")\n",
    "            TCCcount += TCC1\n",
    "            TCG1 = codons.count(\"TCG\")\n",
    "            TCGcount += TCG1\n",
    "            TCT1 = codons.count(\"TCT\")\n",
    "            TCTcount += TCT1\n",
    "            TTA1 = codons.count(\"TTA\")\n",
    "            TTAcount += TTA1\n",
    "            TTT1 = codons.count(\"TTT\")\n",
    "            TTTcount += TTT1\n",
    "            TTC1 = codons.count(\"TTC\")\n",
    "            TTCcount += TTC1\n",
    "            TTG1 = codons.count(\"TTG\")\n",
    "            TTGcount += TTG1\n",
    "            TGT1 = codons.count(\"TGT\")\n",
    "            TGTcount += TGT1\n",
    "            TGC1 = codons.count(\"TGC\")\n",
    "            TGCcount += TGC1\n",
    "            GAA1 = codons.count(\"GAA\")\n",
    "            GAAcount += GAA1\n",
    "            GAT1 = codons.count(\"GAT\")\n",
    "            GATcount += GAT1\n",
    "            GAC1 = codons.count(\"GAC\")\n",
    "            GACcount += GAC1\n",
    "            GAG1 = codons.count(\"GAG\")\n",
    "            GAGcount += GAG1\n",
    "            GTA1 = codons.count(\"GTA\")\n",
    "            GTAcount += GTA1\n",
    "            GTC1 = codons.count(\"GTC\")\n",
    "            GTCcount += GTC1\n",
    "            GTG1 = codons.count(\"GTG\")\n",
    "            GTGcount += GTG1\n",
    "            GTT1 = codons.count(\"GTT\")\n",
    "            GTTcount += GTT1\n",
    "            GCA1 = codons.count(\"GCA\")\n",
    "            GCAcount += GCA1\n",
    "            GCC1 = codons.count(\"GCC\")\n",
    "            GCCcount += GCC1\n",
    "            GCG1 = codons.count(\"GCG\")\n",
    "            GCGcount += GCG1\n",
    "            GCT1 = codons.count(\"GCT\")\n",
    "            GCTcount += GCT1\n",
    "            GGA1 = codons.count(\"GGA\")\n",
    "            GGAcount += GGA1\n",
    "            GGC1 = codons.count(\"GGC\")\n",
    "            GGCcount += GGC1\n",
    "            GGG1 = codons.count(\"GGG\")\n",
    "            GGGcount += GGG1\n",
    "            GGT1 = codons.count(\"GGT\")\n",
    "            GGTcount += GGT1\n",
    "            \n",
    "    \n",
    "    \n",
    "    AAA.append(AAAcount/ (AAAcount+AAGcount)*100)\n",
    "    AAT.append(AATcount/ (AATcount+AACcount)*100)\n",
    "    AAC.append(AACcount/ (AATcount+AACcount)*100)\n",
    "    AAG.append(AAGcount/ (AAAcount+AAGcount)*100)\n",
    "    ACA.append(ACAcount/ (ACAcount+ACTcount+ACCcount+ACGcount)*100)\n",
    "    ACC.append(ACCcount/ (ACAcount+ACTcount+ACCcount+ACGcount)*100)\n",
    "    ACG.append(ACGcount/ (ACAcount+ACTcount+ACCcount+ACGcount)*100)\n",
    "    ACT.append(ACTcount/ (ACAcount+ACTcount+ACCcount+ACGcount)*100)\n",
    "    AGA.append(AGAcount/ (AGAcount+AGGcount)*100)\n",
    "    AGT.append(AGTcount/ (AGTcount+AGCcount)*100)\n",
    "    AGC.append(AGCcount/ (AGTcount+AGCcount)*100)\n",
    "    AGG.append(AGGcount/ (AGAcount+AGGcount)*100)\n",
    "    TAT.append(TATcount/ (TATcount+TACcount)*100)\n",
    "    TAC.append(TACcount/ (TATcount+TACcount)*100)\n",
    "    TTA.append(TTAcount/ (TTAcount+TTGcount)*100)\n",
    "    TTT.append(TTTcount/ (TTTcount+TTCcount)*100)\n",
    "    TTC.append(TTCcount/ (TTTcount+TTCcount)*100)\n",
    "    TTG.append(TTGcount/ (TTAcount+TTGcount)*100)\n",
    "    TCA.append(TCAcount/ (TCAcount+TCTcount+TCGcount+TCCcount)*100)\n",
    "    TCC.append(TCCcount/ (TCAcount+TCTcount+TCGcount+TCCcount)*100)\n",
    "    TCG.append(TCGcount/ (TCAcount+TCTcount+TCGcount+TCCcount)*100)\n",
    "    TCT.append(TCTcount/ (TCAcount+TCTcount+TCGcount+TCCcount)*100)\n",
    "    TGT.append(TGTcount/ (TGTcount+TGCcount)*100)\n",
    "    TGC.append(TGCcount/ (TGTcount+TGCcount)*100)\n",
    "    CAA.append(CAAcount/ (CAAcount+CAGcount)*100)\n",
    "    CAT.append(CATcount/ (CACcount+CATcount)*100)\n",
    "    CAC.append(CACcount/ (CACcount+CATcount)*100)\n",
    "    CAG.append(CAGcount/ (CAAcount+CAGcount)*100)\n",
    "    CTA.append(CTAcount/ (CTAcount+CTCcount+CTCcount+CTTcount)*100)\n",
    "    CTC.append(CTCcount/ (CTAcount+CTCcount+CTCcount+CTTcount)*100)\n",
    "    CTG.append(CTGcount/ (CTAcount+CTCcount+CTCcount+CTTcount)*100)\n",
    "    CTT.append(CTTcount/ (CTAcount+CTCcount+CTCcount+CTTcount)*100)\n",
    "    CCA.append(CCAcount/ (CCAcount+CCTcount+CCCcount+CCGcount)*100)\n",
    "    CCC.append(CCCcount/ (CCAcount+CCTcount+CCCcount+CCGcount)*100)\n",
    "    CCG.append(CCGcount/ (CCAcount+CCTcount+CCCcount+CCGcount)*100)\n",
    "    CCT.append(CCTcount/ (CCAcount+CCTcount+CCCcount+CCGcount)*100)\n",
    "    CGA.append(CGAcount/ (CGAcount+CGTcount+CGCcount+CGGcount)*100)\n",
    "    CGC.append(CGCcount/ (CGAcount+CGTcount+CGCcount+CGGcount)*100)\n",
    "    CGG.append(CGGcount/ (CGAcount+CGTcount+CGCcount+CGGcount)*100)\n",
    "    CGT.append(CGTcount/ (CGAcount+CGTcount+CGCcount+CGGcount)*100)\n",
    "    GAA.append(GAAcount/ (GAAcount+GAGcount)*100)\n",
    "    GAT.append(GATcount/ (GACcount+GATcount)*100)\n",
    "    GAC.append(GACcount/ (GACcount+GATcount)*100)\n",
    "    GAG.append(GAGcount/ (GAAcount+GAGcount)*100)\n",
    "    GTA.append(GTAcount/ (GTAcount+GTTcount+GTCcount+GTGcount)*100)\n",
    "    GTC.append(GTCcount/ (GTAcount+GTTcount+GTCcount+GTGcount)*100)\n",
    "    GTG.append(GTGcount/ (GTAcount+GTTcount+GTCcount+GTGcount)*100)\n",
    "    GTT.append(GTTcount/ (GTAcount+GTTcount+GTCcount+GTGcount)*100)\n",
    "    GCA.append(GCAcount/ (GCAcount+GCTcount+GCCcount+GCGcount)*100)\n",
    "    GCC.append(GCCcount/ (GCAcount+GCTcount+GCCcount+GCGcount)*100)\n",
    "    GCG.append(GCGcount/ (GCAcount+GCTcount+GCCcount+GCGcount)*100)\n",
    "    GCT.append(GCTcount/ (GCAcount+GCTcount+GCCcount+GCGcount)*100)\n",
    "    GGT.append(GGTcount/ (GGAcount+GGTcount+GGCcount+GGGcount)*100)\n",
    "    GGA.append(GGAcount/ (GGAcount+GGTcount+GGCcount+GGGcount)*100)\n",
    "    GGG.append(GGGcount/ (GGAcount+GGTcount+GGCcount+GGGcount)*100)\n",
    "    GGC.append(GGCcount/ (GGAcount+GGTcount+GGCcount+GGGcount)*100)    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saving data to the results/ folder\n"
     ]
    }
   ],
   "source": [
    "df = pd.DataFrame(\n",
    "    data=list(zip(\n",
    "        FileNames,\n",
    "        IntergenomicGC,\n",
    "        AAA,\n",
    "        ACA,\n",
    "        AGA,\n",
    "        TTA,\n",
    "        TCA,\n",
    "        CAA,\n",
    "        CTA,\n",
    "        CCA,\n",
    "        CGA,\n",
    "        GAA,\n",
    "        GTA,\n",
    "        GCA,\n",
    "        GGA,\n",
    "        AAT,\n",
    "        ACT,\n",
    "        AGT,\n",
    "        TAT,\n",
    "        TTT,\n",
    "        TCT,\n",
    "        TGT,\n",
    "        CAT,\n",
    "        CTT,\n",
    "        CCT,\n",
    "        CGT,\n",
    "        GAT,\n",
    "        GTT,\n",
    "        GCT,\n",
    "        GGT,\n",
    "        AAC,\n",
    "        ACC,\n",
    "        AGC, \n",
    "        TAC,\n",
    "        TTC,\n",
    "        TCC,\n",
    "        TGC,\n",
    "        CAC,\n",
    "        CTC,\n",
    "        CCC,\n",
    "        CGC,\n",
    "        GAC,\n",
    "        GTC,\n",
    "        GCC,\n",
    "        GGC,\n",
    "        AAG,\n",
    "        ACG,\n",
    "        AGG,\n",
    "        TTG,\n",
    "        TCG,\n",
    "        CAG,\n",
    "        CTG,\n",
    "        CCG,\n",
    "        CGG,\n",
    "        GAG,\n",
    "        GTG,\n",
    "        GCG,\n",
    "        GGG)\n",
    "    ),\n",
    "    columns=[\n",
    "        \"Name\",\n",
    "        \"IntergenicGC\",\n",
    "        \"AAA\",\n",
    "        \"ACA\",\n",
    "        \"AGA\",\n",
    "        \"TTA\",\n",
    "        \"TCA\",\n",
    "        \"CAA\",\n",
    "        \"CTA\",\n",
    "        \"CCA\",\n",
    "        \"CGA\",\n",
    "        \"GAA\",\n",
    "        \"GTA\",\n",
    "        \"GCA\",\n",
    "        \"GGA\",\n",
    "        \"AAT\",\n",
    "        \"ACT\",\n",
    "        \"AGT\",\n",
    "        \"TAT\",\n",
    "        \"TTT\",\n",
    "        \"TCT\",\n",
    "        \"TGT\",\n",
    "        \"CAT\",\n",
    "        \"CTT\",\n",
    "        \"CCT\",\n",
    "        \"CGT\",\n",
    "        \"GAT\",\n",
    "        \"GTT\",\n",
    "        \"GCT\",\n",
    "        \"GGT\",\n",
    "        \"AAC\",\n",
    "        \"ACC\",\n",
    "        \"AGC\", \n",
    "        \"TAC\",\n",
    "        \"TTC\",\n",
    "        \"TCC\",\n",
    "        \"TGC\",\n",
    "        \"CAC\",\n",
    "        \"CTC\",\n",
    "        \"CCC\",\n",
    "        \"CGC\",\n",
    "        \"GAC\",\n",
    "        \"GTC\",\n",
    "        \"GCC\",\n",
    "        \"GGC\",\n",
    "        \"AAG\",\n",
    "        \"ACG\",\n",
    "        \"AGG\",\n",
    "        \"TTG\",\n",
    "        \"TCG\",\n",
    "        \"CAG\",\n",
    "        \"CTG\",\n",
    "        \"CCG\",\n",
    "        \"CGG\",\n",
    "        \"GAG\",\n",
    "        \"GTG\",\n",
    "        \"GCG\",\n",
    "        \"GGG\"\n",
    "    ],\n",
    "    index=range(1, files_tot+1)\n",
    ")\n",
    "\n",
    "folder = \"results/\"\n",
    "print(f\"Saving data to the {folder} folder\")\n",
    "df.to_csv(f\"{folder}data.csv\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
